[Chan Wei Jie] iP

ChanWeiJie opened the pull request:
## TaskJamie
> "The key is not to prioritize what's on your schedule, but to schedule your priorities" - Stephen Covey [(source)](https://dansilvestre.com/productivity-quotes/)

TaskJamie helps you to schedule your priorities. It's
- easy to learn
- ~~FAST~~ **SUPER FAST** to use
- text-based

All you need to do is,
1. download it from [here](https://github.com/ChanWeiJie/ip/releases/tag/A-Jar)
2. Double click on the duke icon
3. add your tasks
4. let it manage your tasks for you :smiley:

Best part about TaskJamine, _**ITS FREE!!**_

Features :
- [X] Managing tasks (todo)
- [X] Managing dealines
- [X] Managing events 
- [X] Removing tasks
- [X] Finding tasks with a specific word

 Here's the `main` method to start TaskJamie:

```ruby
public static void main(String[] args) {
    Application.launch(Main.class, args);
}
```

Commit:
Add Gradle support

ChanWeiJie committed:
Completed Level 1. Greet, Echo, Exit

ChanWeiJie committed:
Changed name

ChanWeiJie committed:
Completed Level 2. Add & List

ChanWeiJie committed:
Completed Level 3. Mark as done. Also implemented an addtional Task class

ChanWeiJie committed:
Completed level 4 ToDos, Events, Deadlines. Added Todo, Deadline & Event classes

ChanWeiJie committed:
A-TextUiTesting Completed. Added own test case and expected text file

ChanWeiJie committed:
minor changes such as addition and removal of fullstops & spaces

ChanWeiJie committed:
Completed level 5. Handle Errors. added on 4 exception files

ChanWeiJie committed:
Completed level 6, Delete.

ChanWeiJie committed:
added more inputs into input.txt to test functionality of level 5 and level 6

ChanWeiJie committed:
Include new variables to remove magic numbers

ChanWeiJie committed:
Complete level 7 save

ChanWeiJie committed:
commit to change back to branch-level-7

ChanWeiJie committed:
Update duke to save task automatically whenever task list changed

ChanWeiJie committed:
Complete Level 8 Dates and Times

ChanWeiJie committed:
Update addToList method

ChanWeiJie committed:
Merge branch 'branch-Level-7'

ChanWeiJie committed:
add spaces

ChanWeiJie committed:
minor change

ChanWeiJie committed:
Merge branch 'branch-Level-8'

ChanWeiJie committed:
minor change include spaces

ChanWeiJie committed:
Complete A-MoreOOP

Add ability of paths to be OS-independent

Add checks and creations of relevant directories and text file

ChanWeiJie committed:
Complete A-Packages

ChanWeiJie committed:
Minor change inclsive of remove redundant objects

ChanWeiJie committed:
Update A-Packages to put every class under the duke package

ChanWeiJie committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ChanWeiJie committed:
Complete A-Junit

Add TaskTest class

Add ParseTest class

ChanWeiJie committed:
Add Find feature

ChanWeiJie committed:
Add JavaDocs for all classes and public methods, except for getters/setters

ChanWeiJie committed:
Update coding standard

ChanWeiJie committed:
Update Javadocs for all classes

ChanWeiJie committed:
Merge branch 'branch-A-JavaDoc'

ChanWeiJie committed:
Resolve Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/PrintCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/exception/BlankCommandException.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/functionality/Parser.java
#	src/main/java/duke/functionality/Storage.java
#	src/main/java/duke/functionality/TaskList.java

ChanWeiJie committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java

ChanWeiJie committed:
Update Javadocs and Coding Standard for Find feature

ChanWeiJie committed:
Update Javadocs

#TaskList.java and Ui.java updated javadocs.

ChanWeiJie committed:
Update coding standard

ChanWeiJie committed:
Update build.gradle

changed application path to package path

ChanWeiJie committed:
Update test files to use gradle

ChanWeiJie committed:
Merge tag 'A-Gradle'

ChanWeiJie committed:
Add checkstyle.xml and suppressions.xml

Created a new config and checkstyle folder to store both xml files.

Updated the coding standard of respective java files with respect to the output of the xml files.

ChanWeiJie committed:
Merge branch 'branch-A-CheckStyle'

ChanWeiJie committed:
Add GUI for duke

ChanWeiJie committed:
Merge branch 'branch-Level-10'

darshan152 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ * Represents the commands inputted by a user. The Command class cannot be instantiated as it is an abstract class.
+ */
+public abstract class Command {
```
Good use of Abstract Class, was wondering why not use an interface instead? It would save you having to add nulls to the attributes in all your commands.

darshan152 commented on src/main/java/duke/functionality/Parser.java:
```diff
@@ -0,0 +1,164 @@
+            return new PrintCommand();
+
+        } else if (command.equals("todo")) {
+            String description = input.substring(TODO_OFFSET).trim();
```
Good use of static final attributes instead of magic numbers

darshan152 commented on src/main/java/duke/functionality/Parser.java:
```diff
@@ -0,0 +1,164 @@
+
+        } else if (command.equals("deadline")) {
+            String[] inputSlash = input.split("/");
+            String description = inputSlash[0].substring(DEADLINE_OFFSET).trim();
```
Nice usage of `trim()` to check for empty descriptions

darshan152 commented on src/main/java/duke/functionality/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+     * Returns a string which is the collection of all the task in taskList.
+     * @return accumulated tasks.
+     */
+    public String printList() {
```
why not use a toString override? 😄

darshan152 commented on src/main/java/duke/functionality/Ui.java:
```diff
@@ -0,0 +1,57 @@
+ */
+public class Ui {
+    private static final String GREETING = "Hello! I'm TaskJamie\nWhat can i do for you?";
+    private static final String ENDING = "Bye. Hope to see you again soon!";
```
Exit String was already returned in the ExitCommand. Is there a need for it here as well?

darshan152 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+ * and corresponds to a deadline input by a user.
+ */
+public class Deadline extends Task {
+    protected LocalTime time;
```
Why not use private instead? Since there is no need to extend the class anymore

darshan152 reviewed (commented):
LGTM, just some minor suggestions

geetengtan commented on src/main/java/duke/functionality/Parser.java:
```diff
@@ -0,0 +1,164 @@
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
```
Great newline usage to separate imports from different packages

geetengtan commented on src/main/java/duke/functionality/Storage.java:
```diff
@@ -0,0 +1,153 @@
+     * @return a TaskList object representing the saved taskList.
+     * @throws FileNotFoundException if the file is missing.
+     */
+    public TaskList readFileDataAndStoreInList(File file) throws FileNotFoundException {
```
Very descriptive method name 👍

geetengtan commented on src/main/java/duke/functionality/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+     * @param taskNum an indicator to the index of taskList.
+     * @return the specified task to be unmarked.
+     */
+    public Task unMarkTask(int taskNum) {
```
Should it be unmarkTask instead?

geetengtan commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + date.format(DateTimeFormatter.ofPattern("MMM dd yyyy"))
```
Great use of line breaks to improve readability.

geetengtan commented on src/main/java/duke/main/MainWindow.java:
```diff
@@ -0,0 +1,70 @@
+    private void handleUserInput() {
+        String input = userInput.getText();
+        String response = duke.getResponse(input);
+        dialogContainer.getChildren().addAll(
```
Nice usage of line breaks

geetengtan reviewed (commented):
Great work 👍

ChanWeiJie committed:
Update build.gradle

build.gradle : add the implementation groups for mac and linux

Adding these implementation group would allow different system base
to run my jar file.

Let's add all the relevant implementation group statments into
build.gradle.

Refer to this website on why is it done this way
https://se-education.org/guides/tutorials
/gradle.html#creating-a-jar-file

ChanWeiJie committed:
Add assertions

Delete, Mark and Unamrk command : add assertion to index

The tasklist starts from index 1 and not 0.

An assertion on index being greater than 1 allows us to ensure
that the tasklist is being accessed correctly.

Let's put these assertions in the Delete, Mark and Unmark Command
classes.

ChanWeiJie committed:
Update code quality

Current code is working but could use an update in its quality.

Code quality improves understandability and readability.

Let's
* remove obvious comments
* remove magic numbers
* delete dead code such as the UI class
* minimize code duplication
* minimize scope of variables

ChanWeiJie committed:
Merge pull request #1 from ChanWeiJie/branch-A-Assertions

Add assertions

ChanWeiJie committed:
Update code quality

ChanWeiJie committed:
Merge branch 'master' into branch-A-CodeQuality

ChanWeiJie committed:
Merge pull request #2 from ChanWeiJie/branch-A-CodeQuality

Update code quality

ChanWeiJie committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/ChanWeiJie/ip into branch-A-CodeQuality

ChanWeiJie committed:
Merge branch 'branch-A-CodeQuality'

ChanWeiJie committed:
Update build.gradle

ChanWeiJie committed:
Update code quality and exit functionality

ChanWeiJie committed:
Add gradle.yml for continuous integration

remove text ui test file

ChanWeiJie committed:
Add view schedule functionality

ChanWeiJie committed:
Add view schedule functionality

ChanWeiJie committed:
Merge branch 'branch-B-ViewSchedules'

ChanWeiJie committed:
Add Ui.png and update user guide

ChanWeiJie committed:
update README.md

ChanWeiJie committed:
Update README.md

ChanWeiJie committed:
Update Ui.png

ChanWeiJie committed:
Update README.md

ChanWeiJie committed:
Update README.md

ChanWeiJie committed:
Update REDME.md

ChanWeiJie committed:
Update README.md

ChanWeiJie committed:
Update README.md
